### PR TITLE
Fix power and frequency initiailzations

### PR DESCRIPTION
**Pull request by jerryneedell**
Most of the changes were to make this driver compatible with the Arduino driver.
The main changes are to the way the frequency was set - the actual "setter" was never being called and the default frequency (424mHz) remained the actual setting. In addition there was an error in calculation of the frequency setting due to the use of "//" vs "/".

The default baudrate of 10Mhz was causing some "hangs" during reads from the FIFO, even with a featherwing. Lowering it to 5MHz seems to have mad this much more reliable

Several minor changes were made bring the code to closer agreement with the Arduino dirver.

Added an arbitrary 2 second timeout to the send function. It may be changed va a keyword argument. but I wanted to avoid possible infinite loop.

Before these changes, two devices running CP could communicate, but they could not communicate to a device using Arduino. This was due to the frequency mis-configuration. The two CP devices were consistent with eachother, but not with the Arduino Driver.

Also renamed the example to rfm9x_simpletest as is the new custom.

I have tested this with rfm9x feathewings connnected to feather_m0_express boards running teh current CP3.0 master.

### CODE REVIEW

File: adafruit_rfm9x.py [hunk 1]
```diff
@@ -551,12 +563,15 @@ def send(self, data):
...
         # best that can be done right now without interrupts).
-        while not self.tx_done:
-            pass
-        # Clear interrupts.
-        self._write_u8(_RH_RF95_REG_12_IRQ_FLAGS, 0xFF)
+        start = time.monotonic()
+        timed_out = False
+        while not timed_out and not self.tx_done:
+            if (time.monotonic() - start) >= timeout_s:
+                timed_out = True
```
Comment (tannewt) [inline adafruit_rfm9x.py]: Please throw a RuntimeException in the case where tx_done is still false.

File: adafruit_rfm9x.py [hunk 2]
```diff
@@ -523,10 +534,11 @@ def rssi(self):
         # Remember in LoRa mode the payload register changes function to RSSI!
         return self._read_u8(_RH_RF95_REG_1A_PKT_RSSI_VALUE) - 137
 
-    def send(self, data):
+    def send(self, data, timeout_s=2.):
```
Comment (tannewt) [inline adafruit_rfm9x.py]: No `_s`. Just document that its in seconds. This should be our standard for timeouts. (`time.sleep` is an example of the same units.)

File: adafruit_rfm9x.py [hunk 3]
```diff
@@ -576,32 +591,34 @@ def receive(self, timeout_s=0.5, keep_listening=True):
...
         # interrupt supports.
         start = time.monotonic()
-        while not self.rx_done:
+        timed_out = False
+        while not timed_out and not self.rx_done:
             if (time.monotonic() - start) >= timeout_s:
-                return None  # Exceeded timeout.
-        # Clear interrupt.
-        self._write_u8(_RH_RF95_REG_12_IRQ_FLAGS, 0xFF)
+                timed_out = True
```
Comment (tannewt) [inline adafruit_rfm9x.py]: Exception here too and it'll "return" early so you don't need the extra state.

File: adafruit_rfm9x.py [hunk 4]
```diff
@@ -494,6 +498,10 @@ def tx_power(self):
         high power devices (RFM95/96/97/98, high_power=True) or -1 to 14 for low
         power devices. Only integer power levels are actually set (i.e. 12.5
         will result in a value of 12 dBm).
+        The actual maximum setting for high_power=True is 20dBm but for values > 20
+        the PA_BOOST will be enabled  resultiung in ad additonla gain of 3dBm.
```
Comment (tannewt) [inline adafruit_rfm9x.py]: typos

File: adafruit_rfm9x.py (hunk 2 above)
Comment (jerryneedell) [reply adafruit_rfm9x.py]: OK - this was just a holdover from the original code. The change makes sense to me.

File: adafruit_rfm9x.py (hunk 3 above)
Comment (jerryneedell) [reply adafruit_rfm9x.py]: As above, there is still code that needs to execute after the timeout. 
More importantly, I don't think we want an exception here. The typical use of receive it to see if any packets are available. A timeout is normal here.  

We could break this into two parts like it is for Arduino and have an "available" function that the user polls until a packet is received.

File: adafruit_rfm9x.py (hunk 4 above)
Comment (jerryneedell) [reply adafruit_rfm9x.py]: sigh...My fingers are usually way ahead of my brain.

Comment (tannewt) [reply adafruit_rfm9x.py]: Yup, happens to me too. No worries. :-)

File: adafruit_rfm9x.py (hunk 3 above)
Comment (tannewt) [reply adafruit_rfm9x.py]: Ok, returning None is fine with me. I believe we do want an exception for `send` failure above though.

**Comment by jerryneedell** [comment]
I think, hope, I got them all. Let me know if I missed something.
Thanks for the thorough review.

**Pull request merged**